lib/access_import/client_transform.rb

```diff
@@ -0,0 +1,127 @@
+    @familien_rollen = acimport.familien_rollen
+    @plz = acimport.plz
+  end
+
+  def prepare_attributes(personen_rolle)
+    haupt_person = @haupt_personen.find(personen_rolle[:fk_Hauptperson])
+    begleitet, relatives = handle_begleitete(personen_rolle, haupt_person)
+    familien_rolle = @familien_rollen.find(begleitet[:fk_FamilienRolle])
+    land = @laender.find(haupt_person[:fk_Land]) if haupt_person[:fk_Land]
+    plz = @plz.find(haupt_person[:fk_PLZ])
+    sprachen = @sprache_hauptperson.where_person(haupt_person[:pk_Hauptperson])
+    {
+      salutation: salutation(haupt_person[:t_Anrede]),
+      comments: comments(begleitet, personen_rolle, haupt_person),
+      birth_year: birth_year(haupt_person[:d_Geburtsdatum], begleitet[:z_Jahrgang]),
+      entry_year: haupt_person[:d_EintrittCH] && haupt_person[:d_EintrittCH],
+      language_skills_attributes: language_skills_attributes(sprachen),
+      contact_attributes: contact_attributes(haupt_person, plz),
+      relatives_attributes: relatives_attrs(relatives),
+      access_import: access_import(begleitet, haupt_person, personen_rolle,
+        familien_rolle, land, sprachen)
+    }
+  end
+
+  def comments(begleitet, personen_rolle, haupt_person)
+    comments = ''
+    comments += "#{begleitet[:m_Bemerkung]}\n\n" if begleitet[:m_Bemerkung]
+    comments += "#{personen_rolle[:m_Bemerkungen]}\n\n" if personen_rolle[:m_Bemerkungen]
+    comments += "#{haupt_person[:m_Bemerkungen]}\n\n" if haupt_person[:m_Bemerkungen]
+    comments
+  end
+
+  def handle_begleitete(personen_rolle, haupt_person)
+    begleitete = @begleitete.where_personen_rolle(personen_rolle[:pk_PersonenRolle])
+    return [begleitete.first[1], []] if begleitete.size == 1
+    begleitet = begleitete.select do |_key, beg|
+      beg[:fk_FamilienRolle] == 2 # Hauptperson
+    end
+    if begleitet.size == 1
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    begleitet = begleitete.select do |_key, beg|
+      beg[:t_Vorname] == haupt_person[:t_Vorname]
+    end
+    if begleitet.size == 1
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    begleitet = begleitete.select do |_key, beg|
+      haupt_person[:t_Vorname].include? beg[:t_Vorname]
+    end
+    if begleitet.size.positive?
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    # nothing worked, just take first
+    [begleitet.first[1], begleitete.except(begleitet.first[0])]
+  end
+
+  def relatives_attrs(relatives)
+    relatives.map do |_key, relative|
+      [(Time.now.to_f * 1000).to_i, {
+        first_name: relative[:t_Vorname],
+        last_name: relative[:t_Name], relation: relative[:relation],
+        birth_year: relative[:z_Jahrgang] && Date.parse("#{relative[:z_Jahrgang]}-01-01")
+      }]
+    end.to_h
+  end
+
+  def birth_year(geburtsdatum, jahrgang)
+    return geburtsdatum if geburtsdatum
+    Date.parse(jahrgang + '-06-01') if jahrgang
```

Comment:
it just sets the date in the middle of the year. Because where only using the year, but have a full date field.